### PR TITLE
Mark `ARTests` unavailable on Catalyst

### PR DESCRIPTION
**Pull request by dfeinzimer**
Followup to #946

### CODE REVIEW

**Review by zkline101 — COMMENTED** [review]
The problem here is these tests will still get run on mac catalyst. Sadly, marking a test class as unavailable for a platform won't prevent them from running on that platform.

You will need to prevent them from getting run by using compiler directives (`#if`).

**Comment by zkline101** [comment]
I ran into the same situation when adding visionOS support.

File: Tests/ArcGISToolkitTests/ARTests.swift [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 // limitations under the License.
 
-#if !os(visionOS)
+#if !os(visionOS) || !targetEnvironment(macCatalyst)
 import ArcGIS
 import SwiftUI
 import XCTest
 @testable import ArcGISToolkit
 
+@available(macCatalyst, unavailable)
```
Comment (zkline101) [inline Tests/ArcGISToolkitTests/ARTests.swift:21]: This is no longer needed
```suggestion
```

Comment (dfeinzimer) [reply Tests/ArcGISToolkitTests/ARTests.swift:21]: Even with `#if !os(visionOS) || !targetEnvironment(macCatalyst)`, without this in place the tests still show from the Test Navigator in Xcode with Catalyst as the destination and causes errors when you try to run them:

<p align="center">
  
</p>

With `@available(macCatalyst, unavailable)` in place, they still show in Test Navigator but don't run.

File: Tests/ArcGISToolkitTests/ARTests.swift [hunk 2]
```diff
@@ -12,12 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#if !os(visionOS)
+#if !os(visionOS) || !targetEnvironment(macCatalyst)
```
Comment (zkline101) [inline Tests/ArcGISToolkitTests/ARTests.swift:15]:
```suggestion
#if !os(visionOS) && !targetEnvironment(macCatalyst)
```

**Comment by dfeinzimer** [comment]
Closing in reference to #955

**Pull request closed without merging**